### PR TITLE
Rm gradient accumulation on TPU

### PR DESCRIPTION
**Pull request by muellerzr**
Removes gradient accumulation from all examples if on a TPU as there's a major bug within it apparently on the `xla` side. Will investigate further eventually.

Solves #452  and #450

### CODE REVIEW

File: examples/by_feature/gradient_accumulation.py [hunk 1]
```diff
@@ -110,7 +110,10 @@ def training_function(config, args):
     seed = int(config["seed"])
     batch_size = int(config["batch_size"])
     # New Code #
-    gradient_accumulation_steps = int(args.gradient_accumulation_steps)
+    if accelerator.distributed_type != DistributedType.TPU:
+        gradient_accumulation_steps = int(args.gradient_accumulation_steps)
+    else:
+        gradient_accumulation_steps = 1
```
Comment (sgugger) [inline examples/by_feature/gradient_accumulation.py:116]: No, you should raise an error at the beginning if on TPU with a gradient accumulation steps argument passed > 1.

File: examples/complete_nlp_example.py [hunk 2]
```diff
@@ -196,7 +196,8 @@ def collate_fn(examples):
             if args.with_tracking:
                 total_loss += loss.detach().float()
             accelerator.backward(loss)
-            if step % gradient_accumulation_steps == 0:
+            # We currently do not support nor recommend gradient accumulation on the TPU as bugs were found on the XLA side when running our tests
+            if step % gradient_accumulation_steps == 0 or accelerator.distributed_type == DistributedType.TPU:
```
Comment (sgugger) [inline examples/complete_nlp_example.py:200]: Don't need to change here since `gradient_accumulation_steps` will always be 1.

File: examples/nlp_example.py [hunk 3]
```diff
@@ -146,7 +146,8 @@ def training_function(config, args):
             loss = outputs.loss
             loss = loss / gradient_accumulation_steps
             accelerator.backward(loss)
-            if step % gradient_accumulation_steps == 0:
+            # We currently do not support nor recommend gradient accumulation on the TPU as bugs were found on the XLA side when running our tests
+            if step % gradient_accumulation_steps == 0 or accelerator.distributed_type == DistributedType.TPU:
```
Comment (sgugger) [inline examples/nlp_example.py:150]: Same here.

**Review by sgugger — APPROVED** [review]
Thanks for addressing this!
